client/src/redux/reducers/auth.js

```diff
@@ -0,0 +1,52 @@
+import {
+  REGISTER_SUCCESS,
+  REGISTER_FAIL,
+  USER_LOADED,
+  AUTH_ERROR,
+  LOGIN_SUCCESS,
+  LOGIN_FAIL,
+  LOGOUT
+} from '../actions/types';
+
+const initialState = {
+  token: localStorage.getItem('token'),
+  isAuthenticated: null,
+  loading: false,
+  user: null
+};
+
+export default function (state = initialState, action) {
+  const { type, payload } = action;
+
+  switch (type) {
+    case USER_LOADED:
+      return {
+        ...state,
+        isAuthenticated: true,
+        loading: false,
+        user: payload
+      };
+    case REGISTER_SUCCESS:
+    case LOGIN_SUCCESS:
+      localStorage.setItem('token', payload.token);
```

Comment:
I would move this logic to the login action where the login API call is happening so we have all side effects in one place.

One important note about reducers is that they should be pure functions so they should be free of side effects (we can consider the interaction with localStorage a side effect).

The reason for this rule is to simplify reducers, make them very easy to test, and responsible only for one thing: changing the state of the application.

Reply:
Normally, I have seen some developers to follow that way however you are right the aim of the reducers as a pure functions only to change the state. I have changed as your suggestion besides the function which removes the token. I moved them in to actions. Thank you.